include/dem/force_chains_visualization.h

```diff
@@ -0,0 +1,132 @@
+#include <core/auxiliary_math_functions.h>
+#include <core/dem_properties.h>
+
+#include <dem/data_containers.h>
+#include <dem/dem_contact_manager.h>
+#include <dem/dem_solver_parameters.h>
+#include <dem/particle_particle_contact_force.h>
+#include <dem/particle_particle_contact_info.h>
+#include <dem/rolling_resistance_torque_models.h>
+
+#include <deal.II/particles/particle_handler.h>
+
+#include <boost/range/adaptor/map.hpp>
+
+#include <vector>
+
+using namespace dealii;
+using namespace DEM;
+
+#ifndef particles_force_chains_h
+#  define particles_force_chains_h
+
+/**
+ * Base class for the particles force chains contact force models.
+ * This class does not implement any of the models, but ensures that
+ * an interface without template specialization is available. All of the
+ * actual implementation of the models are carried out in the
+ * ParticlesForceChains class which is templated by the contact model
+ * type.
+ */
+template <int dim>
+class ParticlesForceChainsBase
+{
+public:
+  /**
+   * @brief Use a ParticleParticleContactForce object to calculate normal forces between
+   * all touching particles. Stock normal forces and particles position in
+   * vectors.
+   *
+   * @param container_manager The container manager object that contains
+   * containers to modify of contact pair periodic candidates with other
+   * containers with periodic neighbors lists
+   * @param dt DEM time step
+   */
+  virtual void
+  calculate_force_chains(DEMContactManager<dim> &container_manager,
+                         const double            dt) = 0;
```

Comment:
I think this function doesn't need the `dt` as a parameter. 

I would delete this parameter. You use `dt` at line 165 for `update_contact_information` function which is used to update the tangential overlap. The tangential forces are not being used for the forces chains.

 I would just replace the `dt` by `0.` when calling `update_contact_information`.